plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -324,6 +354,80 @@ public CompletableFuture<Collection<Slice>> finish()
         return completedFuture(fragments);
     }
 
+    private Slice writeMergeResult(Slice path, FileDeletion deletion)
+    {
+        Roaring64Bitmap pastDeletionVectors = loadDeletionVectors(Location.of(path.toStringUtf8()));
+
+        Roaring64Bitmap deletedRows = new Roaring64Bitmap();
+        deletedRows.or(pastDeletionVectors);
+        deletedRows.or(deletion.rowsDeletedByDelete());
+        deletedRows.or(deletion.rowsDeletedByUpdate());
+
+        TrinoInputFile inputFile = fileSystem.newInputFile(Location.of(path.toStringUtf8()));
```

Comment:
Do we know size of deletion vector in advance from `io.trino.plugin.deltalake.transactionlog.DeletionVectorEntry#sizeInBytes` or any other metadata ?
If we do, then using it in `newInputFile` would probably save a FS call.

Reply:
This file is a Parquet file, not deletion vector. I think that's possible, but it requires some refactoring. Let me handle in a follow-up.